### PR TITLE
add snippet api definition

### PR DESCRIPTION
**Pull request by dsinghvi**

### CODE REVIEW

File: fern/definition/snippets.yml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      next: 
+        type: optional<string>
+        docs: If present, pass this into the `page` query parameter to load the next set.
+      snippets: list<Snippet>
+
+  EndpointIdentifier: 
+    properties: 
+      endpointPath:
+        type: string
+        docs: The full path of the endpoint (e.g. /snippets/get).
```
Comment (dannysheridan) [inline fern/definition/snippets.yml:64]: Would this be relative path?

File: fern/definition/snippets.yml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        - EndpointNotFound
+    
+    filter: 
+      path: /filter
+      display-name: Filter snippets
+      method: POST
+      request: 
+        name: FilterSnippetsRequest
+        query-parameters:
+          filterId: optional<string>
```
Comment (dannysheridan) [inline fern/definition/snippets.yml:43]: What is this?

File: fern/definition/snippets.yml (hunk 1 above)
Comment (dsinghvi) [reply fern/definition/snippets.yml:113]: renamed to `path`

Comment (dsinghvi) [reply fern/definition/snippets.yml:113]: everything besides the hostname

File: fern/definition/snippets.yml [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      endpointPath:
+        type: string
+        docs: The full path of the endpoint (e.g. /snippets/get). 
+      endpointMethod: 
+        type: EndpointMethod 
+
+  SnippetId: string
+
+  EndpointMethod: 
+    enum:
```
[the remark this answers is not in the document]
Comment (dsinghvi) [reply fern/definition/snippets.yml:120]: can add

File: fern/definition/snippets.yml (hunk 2 above)
Comment (dsinghvi) [reply fern/definition/snippets.yml:43]: its a way to do paging across filter queries

**Pull request closed without merging**